### PR TITLE
feat: replay data services

### PR DESCRIPTION
**Pull request by doudou**
Depends on:
- [ ] https://github.com/rock-core/tools-syskit/pull/408

### CODE REVIEW

File: test/task_streams_test.rb [hunk 1]
```diff
@@ -256,5 +256,75 @@ def validate_stream_name(object_name, stream_name)
...
+
+            it "returns a service-oriented TaskStreams object" do
+                srv_m = Syskit::DataService.new_submodel do
+                    output_port "p", "/int32_t"
+                end
+                srv_streams = @subject.as_data_service(srv_m, "p" => "porta")
+                assert_equal "task.porta", srv_streams.p_port.name
+            end
+
+            it "enumerates the mapped ports" do
```
Comment (wvmcastro) [inline test/task_streams_test.rb:300]: It bothers me that the test proposes to test it enumerates portS but in the test body it is testing against only one

**Review by wvmcastro — COMMENTED** [review]
I think I understand what you have done here, but I can't connect how it solves the camera log replay

File: test/task_streams_test.rb (hunk 1 above)
Comment (doudou) [reply test/task_streams_test.rb:300]: Fixed.

**Comment by doudou** [comment]
> I think I understand what you have done here, but I can't connect how it solves the camera log replay

One of the major current issues with replay scripts is that you have to pretend that the replayed tasks is what it was. For device drivers, it means it has to match what a device driver does, which for instance mean having a device associated with it. 

In other cases, the issue is with dynamic services. If the port was part of a dynamic service, you need to update the model to match. A pain.

This makes sure we can easily replay some ports by telling Syskit they are service X and then injecting them in compositions that accept service X. No more pretending.

**Review by wvmcastro — APPROVED** [review]

**Pull request merged**